extensions/standard-processors/processors/SplitRecord.h

```diff
@@ -0,0 +1,86 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "core/Annotation.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "core/ProcessSessionFactory.h"
+#include "core/PropertyDefinition.h"
+#include "core/PropertyDefinitionBuilder.h"
+#include "core/RelationshipDefinition.h"
+#include "core/logging/Logger.h"
+#include "controllers/RecordSetReader.h"
+#include "controllers/RecordSetWriter.h"
+#include "core/AbstractProcessor.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+class SplitRecord final : public core::AbstractProcessor<SplitRecord> {
+ public:
+  using core::AbstractProcessor<SplitRecord>::AbstractProcessor;
+
+  EXTENSIONAPI static constexpr const char* Description = "Splits up an input FlowFile that is in a record-oriented data format into multiple smaller FlowFiles";
+
+  EXTENSIONAPI static constexpr auto RecordReader = core::PropertyDefinitionBuilder<>::createProperty("Record Reader")
+      .withDescription("Specifies the Controller Service to use for reading incoming data")
+      .isRequired(true)
+      .withAllowedTypes<minifi::core::RecordSetReader>()
+      .build();
+  EXTENSIONAPI static constexpr auto RecordWriter = core::PropertyDefinitionBuilder<>::createProperty("Record Writer")
+      .withDescription("Specifies the Controller Service to use for writing out the records")
+      .isRequired(true)
+      .withAllowedTypes<minifi::core::RecordSetWriter>()
+      .build();
+  EXTENSIONAPI static constexpr auto RecordsPerSplit = core::PropertyDefinitionBuilder<>::createProperty("Records Per Split")
+      .withDescription("Specifies how many records should be written to each 'split' or 'segment' FlowFile")
+      .isRequired(true)
+      .supportsExpressionLanguage(true)
+      .build();
+
+  EXTENSIONAPI static constexpr auto Properties = std::to_array<core::PropertyReference>({
+      RecordReader,
+      RecordWriter,
+      RecordsPerSplit
+  });
+
+  EXTENSIONAPI static constexpr auto Failure = core::RelationshipDefinition{"failure",
+      "If a FlowFile cannot be transformed from the configured input format to the configured output format, the unchanged FlowFile will be routed to this relationship."};
+  EXTENSIONAPI static constexpr auto Splits = core::RelationshipDefinition{"splits",
+      "The individual 'segments' of the original FlowFile will be routed to this relationship."};
+  EXTENSIONAPI static constexpr auto Original = core::RelationshipDefinition{"original",
+      "Upon successfully splitting an input FlowFile, the original FlowFile will be sent to this relationship."};
+  EXTENSIONAPI static constexpr auto Relationships = std::array{Failure, Splits, Original};
+
```

Comment:
Output attribute documentation can also be auto-generated (search for `OutputAttributeDefinition` for examples).